scripts/mantidpython.py

```diff
@@ -3,45 +3,156 @@
 import re
 import subprocess
 import sys
+import tempfile
 
-mantidRegExp=re.compile(r'/opt/.antid.*/bin')
-def getMantidLoc(line):
+mantidRegExp = re.compile(r'/opt/.antid.*/bin')
+
+MANDID_VERSION = 'MANTID_VERSION='
+CONDA_NAME = 'CONDA_ENV='
+NSD_CONDA_WRAP = '/SNS/users/wzz/Projects/AutoReduction/nsd-app-wrap/nsd-conda-wrap.sh'
+
+mantid_version_dict = {'nightly': "/opt/mantidnightly/bin",
+                       'stable': "/opt/Mantid/bin"}
+
+
+def get_mantid_loc(line):
+    """Get Mantid bin or mantid python location
+
+    MANTID_VERSION='nightly'  # returns "/opt/mantidnightly/bin"
+    MANTID_VERSION='60'       # returns "/opt/mantid60/bin"
+    MANTID_VERSION='stable'   # returns "/opt/Mantid/bin"
+
+    CONDA_ENV='sasview'
+    CONDA_ENV='imaging'
+
+    Parameters
+    ----------
+    line: str
+        one line in auto reduction file
+
+    Returns
+    -------
+    ~tuple
+        mantid python location (None for not found), conda env name (None for not found)
+
+    """
     line = line.strip()
-    if not line.startswith("sys.path"):
-        return None
-    mantidversion = mantidRegExp.findall(line)
-    if len(mantidversion)==1:
-        return mantidversion[0]
+    if line.startswith("sys.path"):
+        # backward compatible: sys.path.append or sys.path.insert
+        mantidversion = mantidRegExp.findall(line)
+        print('Mantid version: {}'.format(mantidversion))
+        if len(mantidversion) == 1:
+            return mantidversion[0], None
+
+    elif line.startswith(MANDID_VERSION):
+        #  specify mantid version as: 50, 60, nightly, stable
+        mantid_version = line.split(MANDID_VERSION)[1].split()[0].replace('"', '').replace("'", '')
+        if mantid_version in mantid_version_dict:
+            # label: nightly, stable
+            return mantid_version_dict[mantid_version], None
+        else:
+            # version number
+            return '/opt/mantid{}/bin'.format(mantid_version), None
+
+    elif line.startswith(CONDA_NAME):
+        # CONDA name
+        conda_name = line.split(CONDA_NAME)[1].split()[0].replace("'", '').replace('"', '')
+        return None, conda_name
+
+    return None, None
+
+
+def main():
+    # parse argument 1: reduction script
+    reductionScript = open(sys.argv[1], 'r')
```

Comment:
This would be better with a context manager which would close the file when you are done
```suggestion
    with open(sys.argv[1], 'r') as reductionScript:
        reduction_commands = generate_subprocess_command(reductionScript, sys.argv[2:], True)
```